pkg/nodehandler/controller.go

```diff
@@ -229,9 +231,47 @@ func addNodeRemovalLabel(csibmnode *nodecrd.Node) {
 	if csibmnode.Labels == nil {
 		csibmnode.Labels = map[string]string{}
 	}
-	csibmnode.Labels[nodeRemovalTaintKey] = nodeRemovalTaintValue
+	csibmnode.Labels[rTaint.Key] = rTaint.Value
 }
 
 func deleteNodeRemovalLabel(csibmnode *nodecrd.Node) {
-	delete(csibmnode.Labels, nodeRemovalTaintKey)
+	delete(csibmnode.Labels, rTaint.Key)
+}
+
+func (c *Controller) handleNodeMaintenance(ctx context.Context, nodes []corev1.Node) error {
+	for i, node := range nodes {
+		if hasTaint(&nodes[i], mTaint) {
+			if err := c.deleteCSIPods(ctx, node.Name); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *Controller) deleteCSIPods(ctx context.Context, nodeName string) error {
+	c.log.Info(fmt.Sprintf("Starting to delete CSI pods for on node %s", nodeName))
+
+	fieldSelector := fields.SelectorFromSet(map[string]string{"spec.nodeName": nodeName})
+	labelSelector := labels.SelectorFromSet(common.ConstructLabelAppMap())
+
+	var pods corev1.PodList
+	err := c.client.List(ctx, &pods, &client.ListOptions{FieldSelector: fieldSelector, LabelSelector: labelSelector})
+	if err != nil {
+		return err
+	}
+
+	if len(pods.Items) == 0 {
+		c.log.Info(fmt.Sprintf("There are no CSI pods on the node %s", nodeName))
+	}
+
+	for _, pod := range pods.Items {
+		c.log.Info(fmt.Sprintf("Going to remove %s", pod.Name))
+		// Remove Pod
+		if err := c.client.Delete(ctx, pod.DeepCopy()); err != nil {
```

Comment:
Could we try to delete all pods before returning error?

Reply:
What did mean that we need to ignore errors for pods until loop is over?

Reply:
Done.